### DISCUSSION TITLE
🐞 TUI appears frozen when pulling large image and using `Terminal`

### DISCUSSION THREAD

**Discussion by levlaz** [dagger/dagger#12268] [Help]
### What is the issue?

This was reported by Nipuna in discord[1]

The terminal command seems to kick into "pre" terminal mode before the from image is pulled. So while the image is being pulled the TUI freezes on my screen. 

I expected to have access to the TUI until at least the image for the corresponding container is pulled so I can watch that progress and know that it's being pulled. Pulling a large image can take time and it feels like everything is frozen while it does.

[1]  https://discord.com/channels/707636530424053791/1120503349599543376/1261442362220220477 

### Dagger version

dagger v0.12.0 (registry.dagger.io/engine) linux/amd64

### Steps to reproduce

This repo shows an example, but it should be the same with any large image that you don't already have on your machine. 

```
dagger call -m gitea.com/levlaz/repro-terminal-freeze debug 
```

Note: once you pull this image its hard to reproduce the error because the image being locally available makes things very fast.

**Comment by samalba** [comment]
Confirmed, I can repro on latest v0.12.4

cc @vito
